redfish-core/include/utils/name_utils.hpp

```diff
@@ -0,0 +1,79 @@
+
+#pragma once
+#include <async_resp.hpp>
+
+#include <algorithm>
+#include <string>
+#include <variant>
+#include <vector>
+
+namespace redfish
+{
+namespace name_util
+{
+
+/**
+ * @brief Populate the collection "Members" from a GetSubTreePaths search of
+ *        inventory
+ *
+ * @param[i,o] asyncResp  Async response object
+ * @param[i]   path       D-bus object path to find the pretty name
+ * @param[i]   services   List of services to exporting the D-bus object path
+ * @param[i]   namePath   Json pointer to the name field to update.
+ *
+ * @return void
+ */
+inline void getPrettyName(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& path,
+    const std::vector<std::pair<std::string, std::vector<std::string>>>&
+        services,
+    const nlohmann::json::json_pointer& namePath)
+{
+    BMCWEB_LOG_DEBUG("Get PrettyName for: {}", path);
+
+    // Ensure we only got one service back
+    if (services.size() != 1)
+    {
+        BMCWEB_LOG_ERROR("Invalid Service Size  {}", services.size());
+        if (asyncResp)
+        {
+            messages::internalError(asyncResp->res);
+        }
+        return;
+    }
```

Comment:
In 1060,  the only first service is used.
Should it be done too for 1110?

Also, I think the number services check should be done by the caller, not inside getPrettyName()?

Reply:
Thank you for identifying it. We will surely address this as part of new JIRA item.